crates/autopilot/src/database/auction.rs

```diff
@@ -73,24 +73,18 @@ impl Postgres {
         sqlx::query("SET TRANSACTION ISOLATION LEVEL REPEATABLE READ")
             .execute(ex.deref_mut())
             .await?;
-        let orders: Vec<Order> = database::orders::solvable_orders(&mut ex, min_valid_to as i64)
-            .map(|result| match result {
-                Ok(order) => full_order_into_model_order(order),
-                Err(err) => Err(anyhow::Error::from(err)),
-            })
-            .try_collect()
-            .await?;
+        let orders: HashMap<domain::OrderUid, Order> =
+            database::orders::solvable_orders(&mut ex, min_valid_to as i64)
```

Comment:
Please, don't cast an `u32` to an `i64` like that, use better `try_from()`. See https://github.com/cowprotocol/services/pull/2828#discussion_r1689617078

Reply:
Why? u32 can be safely casted to i64, can't it?

Reply:
Could even be an `.into()` since any `u32` fits into `i64`.

Reply:
oh yep, you are right, it is `u32` not `u64` :ok_hand: nevertheless, I would then use `from` for the reason that if in the future the type changes, it doesn't create problems since the `from()` would complain, but the `as` wouldn't (I do not like `as` casting for this precise reason).